docs/Private-IP-Configuration.md

```diff
@@ -0,0 +1,99 @@
+4. Select “Direction of traffic” as Ingress & “Action on match” as Allow.
+5. Complete Firewall-Rule creation by providing other necessary information. Refer to GCP’s [Configuring-Firewall](https://cloud.google.com/filestore/docs/configuring-firewall) documentation for more details.
+
+![Firewall Rule Details](_static/firewall_rule_details.png?raw=true "Firewall Rule Details")
+
+## Configure NAT & Router
+1. Open GCP’s Create a NAT gateway page.
+2. Provide name, region.
+3. For the “Network” drop-down, select the network that we created earlier.
+4. For the “Router” drop-down, EITHER select pre-created router OR click on “create new router”.
+<br> a. Complete router creation by providing name, description & region. Refer to GCP’s [Create-Router](https://cloud.google.com/network-connectivity/docs/router/how-to/create-router-vpc-on-premises-network) documentation for more details.
+5. Complete NAT gateway creation by providing required details. Refer to GCP’s [Create-NAT-Gateway](https://cloud.google.com/nat/docs/set-up-manage-network-address-translation) documentation for more details.
+
+![Router Details](_static/router_details.png?raw=true "Router Details")
+
+![NAT Gateways](_static/nat_gateways.png?raw=true "NAT Gateways")
+
+## Sample commands to trigger dataflow pipeline execution using above options
+
+Following section showcases how VPC-parameters can be given as CLI inputs to weather-mv dataflow pipeline.
+
+```bash
+weather-mv --uris "gs://$STORAGE_BUCKET/*.nc"
+           --output_table "$HOST_PROJECT_ID.$DATASET_ID.$TABLE_ID"
+           --temp_location "gs://$STORAGE_BUCKET/tmp"
+           --runner DataflowRunner
+           --project $HOST_PROJECT_ID
+           --region $REGION_NAME
+           --no_use_public_ips 
+           --network=$NETWORK_NAME
+           --subnetwork=regions/$REGION_NAME/subnetworks/$SUBNETWORK_NAME
+```
+
+Replace the following:
+
+- STORAGE_BUCKET: the storage bucket, e.g. bucket_58231
+- HOST_PROJECT_ID: the host project ID, e.g. weather_tools
+- DATASET_ID: the name of dataset, e.g. weather_mv_ds
+- TABLE_ID: the name of table, e.g. tbl_2017_01 
+- REGION_NAME: the regional endpoint of your Dataflow job, e.g. us-central1
+- NETWORK_NAME: the name of your Compute Engine network, e.g. dataflow
+  - Provide network_name same as what we created in Step-1.
+- SUBNETWORK_NAME: the name of your Compute Engine subnetwork, e.g. private
+  - Provide a subnetwork_name same as what we created in Step-1.
+
+Alternatively, you may also execute following command,
+
+```bash
+weather-mv --uris "gs://$STORAGE_BUCKET/*.nc"
+           --output_table "$HOST_PROJECT_ID.$DATASET_ID.$TABLE_ID"
+           --temp_location "gs://$STORAGE_BUCKET/tmp"
+           --runner DataflowRunner
+           --project $HOST_PROJECT_ID
+           --region $REGION_NAME
+           --no_use_public_ips
+           –subnetwork=https://www.googleapis.com/compute/v1/projects/$HOST_PROJECT_ID/regions/$REGION_NAME/subnetworks/$SUBNETWORK_NAME
```

Comment:
Should subnetwork have two dashes? `--`?

Reply:
Said changes implemented.